spec/compliance_data_spec.rb

```diff
@@ -0,0 +1,135 @@
+require 'spec_helper'
+require 'stub_classes'
+
+describe ComplianceData do
+  describe '#initialize' do
+    it 'returns a correctly constructed ComplianceData' do
+      data = ComplianceData.new StubClasses::StubSettings
+      expect(data).to be_an_instance_of ComplianceData
+      expect(data.settings).to eq StubClasses::StubSettings
+    end
+
+    it 'raises an argument error if settings are not correct' do
+      expect { ComplianceData.new nil }.to raise_error 'ArgumentError'
+    end
+  end
+
+  describe '#base_name' do
+    it 'returns correct output if given good input' do
+      @data = ComplianceData.new StubClasses::StubSettings
```

Comment:
Minor: I'd make these regular variables rather than instance variables, since they don't need to be. Ditto below.